src/IECore/MeshAlgo.cpp

```diff
@@ -16,6 +17,169 @@ using namespace Imath;
 
 namespace
 {
+class DeleteFlaggedUniformFunctor
+{
+	public:
+		typedef DataPtr ReturnType;
+
+		DeleteFlaggedUniformFunctor( ConstIntVectorDataPtr flagData ) : m_flagData( flagData )
+		{
+		}
+
+		template<typename T>
+		ReturnType operator()( const T *data )
+		{
+			const typename T::ValueType &inputs = data->readable();
+			const std::vector<int> &flags = m_flagData->readable();
+
+			T *filteredResultData = new T();
```

Comment:
I can't remember what happens here, but should this be `T::Ptr` to increment the refcount for the data?